components/director/pkg/http/request.go

```diff
@@ -0,0 +1,62 @@
+package http
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/pkg/apperrors"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+func RequestWithCredentials(ctx context.Context, client *http.Client, url string, auth *model.Auth) (*http.Response, error) {
```

Comment:
Since those functions `RequestWithCredentials` and `RequestWithoutCredentials` are in a generic context now `httpuntil` and not in the fetch request I think it makes sense to name them:

`GetRequestWithCredentials` and `GetRequestWithoutCredentials`